VideoLocker/src/main/java/org/edx/mobile/user/Account.java

```diff
@@ -0,0 +1,201 @@
+package org.edx.mobile.user;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class Account {
+    @SerializedName("username")
+    @NonNull
```

Comment:
Sounds good!
> On 28-Sep-2015, at 11:49 pm, Brian Guertin <notifications@github.com> wrote:
> 
> In VideoLocker/src/main/java/org/edx/mobile/user/Account.java <https://github.com/edx/edx-app-android/pull/372#discussion_r40589853>:
> 
> > @@ -0,0 +1,201 @@
> > +package org.edx.mobile.user;
> > +
> > +import android.support.annotation.NonNull;
> > +import android.support.annotation.Nullable;
> > +
> > +import com.google.gson.annotations.SerializedName;
> > +
> > +import java.util.ArrayList;
> > +import java.util.Date;
> > +import java.util.List;
> > +
> > +public class Account {
> > +    @SerializedName("username")
> > +    @NonNull
> Silent-failures (like using a blank string when we expected to get a string from the API) are a worse problem then a crashing app, because we won't know that there is anything wrong.
> 
> What if we configured Gson to fail if @NonNull field has a null value? Then it can be treated at runtime like any other deserialization error.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/edx/edx-app-android/pull/372/files#r40589853>.
>